src/utilities/make_state_vector_file.py

```diff
@@ -103,32 +114,34 @@ def make_state_vector_file(
     lat_min_inv_domain = np.max([lat_min - buffer_deg, minLat_allowed])
     lat_max_inv_domain = np.min([lat_max + buffer_deg, maxLat_allowed])
 
-    # Subset inversion domain using land cover file
+    # Subset inversion domain for land cover and hemco diagnostics fields
     lc = lc.isel(lon=lc.lon >= lon_min_inv_domain, lat=lc.lat >= lat_min_inv_domain)
     lc = lc.isel(lon=lc.lon <= lon_max_inv_domain, lat=lc.lat <= lat_max_inv_domain)
+    hd = hd.isel(lon=hd.lon >= lon_min_inv_domain, lat=hd.lat >= lat_min_inv_domain)
+    hd = hd.isel(lon=hd.lon <= lon_max_inv_domain, lat=hd.lat <= lat_max_inv_domain)
 
-    # Replace all values with NaN (to be filled later)
+    # Initialize state vector from land cover, replacing all values with NaN (to be filled later)
     statevector = lc.where(lc == -9999.0)
 
     # Set pixels in buffer areas to 0
     statevector[:, (statevector.lon < lon_min) | (statevector.lon > lon_max)] = 0
     statevector[(statevector.lat < lat_min) | (statevector.lat > lat_max), :] = 0
 
-    # Also set pixels over water to 0
+    # Also set pixels over water to 0, unless there are offshore emissions
     if land_threshold:
-        # Where there is no land, replace with 0
-        land = lc.where(lc > land_threshold)
+        # Where there is neither land nor emissions, replace with 0
+        land = lc.where((lc > land_threshold) | (hd > emis_threshold))
         statevector.values[land.isnull().values] = 0
```

Comment:
I may be confused what is happening here, but why don't we just set these values to -9999 here rather than a few lines down?

Reply:
Good point, I made this change and tested on Cannon. Seems to work as expected